builtin/logical/pki/acme_state.go

```diff
@@ -146,6 +146,8 @@ type acmeAccount struct {
 	Contact              []string          `json:"contact"`
 	TermsOfServiceAgreed bool              `json:"termsOfServiceAgreed"`
 	Jwk                  []byte            `json:"jwk"`
+	LastAccessDate       time.Time         `json:"last_access_date"`
+	MarkedAsRevokedDate  *time.Time        `json:"marked_as_revoked_date"`
```

Comment:
Is there any real value of differentiating between `nil` and `IsZero()` for `MarkedAsRevokedDate` to be a pointer?

Reply:
The value that I saw was that it is far easier to remember when coding to check for nil value than for "0".  It feels awkward because jan 1, 1970 is a totally valid date (and if something was revoked then, we'd totally want to delete it, so that's a bad error).